modules/dataframe/src/properties/filterlistproperty.cpp

```diff
@@ -0,0 +1,207 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/dataframe/properties/filterlistproperty.h>
+#include <inviwo/core/properties/boolcompositeproperty.h>
+#include <inviwo/core/properties/stringproperty.h>
+#include <inviwo/core/properties/boolproperty.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/minmaxproperty.h>
+#include <inviwo/core/properties/optionproperty.h>
+
+#include <inviwo/dataframe/util/filters.h>
+
+#include <limits>
+#include <regex>
+#include <functional>
+#include <algorithm>
+
+namespace inviwo {
+
+const std::string FilterListProperty::classIdentifier = "org.inviwo.FilterListProperty";
+std::string FilterListProperty::getClassIdentifier() const { return classIdentifier; }
+
+FilterListProperty::FilterListProperty(std::string_view identifier, std::string_view displayName,
+                                       bool supportsFilterOnHeader, size_t maxNumberOfElements,
+                                       ListPropertyUIFlags uiFlags,
+                                       InvalidationLevel invalidationLevel,
+                                       PropertySemantics semantics)
+    : ListProperty(identifier, displayName, maxNumberOfElements, uiFlags, invalidationLevel,
+                   semantics) {
+
+    auto onHeaderProp = [enable = supportsFilterOnHeader]() {
+        auto p = std::make_unique<BoolProperty>("filterOnHeader", "Filter on Header", enable);
+        p->setVisible(enable);
+        return p;
+    };
+
+    {
+        auto rowBegin = std::make_unique<BoolCompositeProperty>("rowBegin", "Row Begin", true);
+        rowBegin->addProperty(onHeaderProp());
+        rowBegin->addProperty(std::make_unique<StringProperty>("match", "Matching String", ""));
+
+        addPrefab(std::move(rowBegin));
+    }
+    {
+        auto lineRange = std::make_unique<BoolCompositeProperty>("lineRange", "Line Range", true);
```

Comment:
We use “row” for most other cases. We should make it consistent

Reply:
Because here it actually refers to the line numbers in the file and not the index of the data items (rows).